static/src/javascripts/projects/commercial/modules/dfp/lazy-load.ts

```diff
@@ -33,14 +36,36 @@ const onIntersect = (
 		(advert) => !advertIds.includes(advert.id),
 	);
 };
+/**
+ * return a random number between 20 (inclusive) and 420 (non-inclusive)
+ * @returns number
+ */
+const getRandomLazyLoadMargin = (): number => {
+	const min = 20;
+	const max = 420;
+	return Math.floor(Math.random() * (max - min)) + min;
+};
 
-const getObserver = once(() =>
-	Promise.resolve(
+const getObserver = once(() => {
+	const inLazyLoadMarginTestVariant = isInVariantSynchronous(
+		commercialLazyLoadMargin,
+		'variant',
+	);
+	let rootMargin;
+	if (inLazyLoadMarginTestVariant) {
+		const margin = getRandomLazyLoadMargin();
+		const eventTimer = EventTimer.get();
+		eventTimer.setProperty('lazyLoadMarginPercent', margin);
+		rootMargin = `${margin}%`;
```

Comment:
Should this also be ```rootMargin = `${margin}% 0px;` ``` ? The non-variant version seems to have vertical and horizontal settings.

Reply:
Good point - I was wondering about this. I _think_ since intersection observer margins work like [css margins](https://developer.mozilla.org/en-US/docs/Web/CSS/margin), using just the one value would load the ad when it's `margin` % from the viewport vertically or horizontally. As far as I know we don't have any pages with horizontal scrolling so I was thinking it wouldn't make too much of a difference. I'll add the horizontal margin to make the intent a bit more clear.